main.tf

```diff
@@ -54,7 +54,7 @@ resource "aws_cloudwatch_event_target" "asg_downscale_scheduler_target" {
 
 resource "aws_cloudwatch_event_rule" "asg_upscale_scheduler_event" {
   count               = var.asg_scheduler != null ? 1 : 0
-  name                = "asg-scheduler-upsacale-event-${random_id.this.id}"
+  name                = "asg-scheduler-upsacale-event-${random_id.this[0].id}"
```

Comment:
I just realized there's a typo in here. upascale instead of upscale ahah

Reply:
Good catch!